ra/ra.go

```diff
@@ -2353,7 +2353,14 @@ func (ra *RegistrationAuthorityImpl) checkOrderNames(names []string) error {
 // or the certificate is expired, it returns berrors.NotFoundError.
 func (ra *RegistrationAuthorityImpl) GenerateOCSP(ctx context.Context, req *rapb.GenerateOCSPRequest) (*capb.OCSPResponse, error) {
 	status, err := ra.SA.GetCertificateStatus(ctx, &sapb.Serial{Serial: req.Serial})
-	if err != nil {
+	if errors.Is(err, berrors.NotFound) {
+		_, err := ra.SA.GetSerialMetadata(ctx, &sapb.Serial{Serial: req.Serial})
+		if errors.Is(err, berrors.NotFound) {
+			return nil, berrors.UnknownSerialError()
+		} else {
```

Comment:
We don't have to do an inner check for a `errIncompleteRequest` because that's already performed above inside the call to `GetCertificateStatus` so it's not worth it again? If `GetSerialMetadata` returns `errIncompleteRequest` or `invalid certificate serial %s", req.Serial` we'll be returning `berrors.NotFoundError` instead of the more informative inner error, is that fine?

Reply:
Yep. Also in general we don't check for `errIncompleteRequest` specifically because it's just a variety of "internal server error"; we messed up somehow.

You do make a valid point that there may be some internal problem in our GetSerialMetadata request that causes all requests to error with something other than NotFound, and we would not see it because it would be returning "not found" to match the GetCertificateStatus request. But I think that's low probability and low impact, so it's okay. Plumbing through or logging the secondary error would be kinda messy.